### PR TITLE
Move coinbase to header

### PR DESCRIPTION
**Pull request by lucafavatella**
Delivers https://www.pivotaltracker.com/story/show/157214052
Respective specs PR: https://github.com/aeternity/protocol/pull/66

(For easing review, commit "Add mining account pubkey to header" is the same - rebased - as in past RFC PR #1165 )

TODO:
* [x] Investigate and fix "aec_conductor_tests:block_candidate_test_/0: Get block candidate" unit test failure
* [x] Investigate integration test `make test` failures
* [x] Ensure full CI run passes
* [x] Ensure Python UAT tests pass - see [auxiliary CI workflow running on 05636c8e that is equivalent to 5a05048b](https://circleci.com/workflow-run/b9e41357-c84b-4ba7-94d0-523b8117face) UPDATE: [Python UAT tests job passed](https://circleci.com/gh/lucafavatella/epoch/3802)
* [x] Ensure system tests pass
  * Auxiliary CI job https://circleci.com/gh/lucafavatella/epoch/3812 on d8284d54 that is equivalent to dd08cc1a (I cherry-picked system test fix 3d2e8257 from master) UPDATE: [Passed except `long_chain.stay_in_sync`](https://3812-110719633-gh.circle-artifacts.com/0/home/circleci/epoch/system_test/logs/ct_run.epoch_ct@localhost.2018-05-22_13.38.01/extras.system_test.logs/run.2018-05-22_13.38.01/suite.log.html) - failing independently from this PR
* [x] Double check test coverage UPDATE: I see unit test coverage of block mining reward to miner in `aetx_tests`
* [x] Release notes
* [x] Bump DB folder for environments
* [x] Check if any updates to https://github.com/aeternity/protocol needed further to PR https://github.com/aeternity/protocol/pull/66 UPDATE: Content of PR 66 updated, list of reviewers updated.
* [x] Check if any updates to https://github.com/aeternity/epoch/wiki needed
  * UPDATE: Refreshed reference to specs (wiki commit 2bf790ba - [suboptimal comparison UI link](https://github.com/aeternity/epoch/wiki/Home/_compare/6a39764df583f10aa172d6fcce47bdcac11d9e14...2bf790ba99fd3d3e2c7d1b5bd262cc647a25217e)
  * UPDATE: Updated HTTP examples in wiki ([changes](https://github.com/aeternity/epoch/wiki/HTTP-Client-API:-get-transactions-list-from-a-block-range/_compare/c1a9193c0d69a2b176a6f33cd1544c87c6626a19...8d3e676e1e950cfaf28bb62dde372445aba853cf))
* [ ] (Probably) Re-rebase on current master
* [ ] Squash-and-merge with commit message:
> Move coinbase to header
>
> The mining reward is now applied after the application of transactions - it used to be applied before (because of coinbase tx as first tx).
>
> No API for querying the blocks mined by a certain miner is added - it used to be possible to query that by piggybacking on the txs-per-account query.

### CODE REVIEW

**Comment by lucafavatella** [comment]
There is no need to rebase to master (nothing relevant got merged).
I let CI run system tests though I consider this ready for review.

File: apps/aetx/src/aetx.erl [hunk 1]
```diff
@@ -174,7 +171,7 @@ signers(#aetx{ cb = CB, tx = Tx }, Trees) ->
             ConsensusVersion :: non_neg_integer()) ->
     {ok, NewTrees :: aec_trees:trees()} | {error, Reason :: term()}.
 check(#aetx{ cb = CB, tx = Tx } = O, Trees, Height, ConsensusVersion) ->
```
Comment (hanssv) [inline apps/aetx/src/aetx.erl]: `O` becomes unused here...

File: apps/aecore/test/aec_trees_tests.erl [hunk 2]
```diff
@@ -46,24 +47,40 @@ signatures_check_test_() ->
...
             {ok, ApprovedTxs, _Trees} =
-                ?TEST_MODULE:apply_signed_txs(MalformedTxs, aec_trees:new(), 1,
+                ?TEST_MODULE:apply_signed_txs(?MINER_PUBKEY, MalformedTxs, aec_trees:new(), 1,
                                           ?PROTOCOL_VERSION),
             ?assertEqual([], ApprovedTxs),
             ok
         end}
      ]}.
 
+make_a_spend_tx(Sender, Recipient) ->
```
Comment (zp-sd) [inline apps/aecore/test/aec_trees_tests.erl]: 💅 maybe just `make_spend_tx`?

**Comment by lucafavatella** [comment]
I will address comments in separate PR so not to delay merging of most of the code.

**Pull request merged**